ProcessesApi/V1/Services/SoleToJointService.cs

```diff
@@ -131,6 +131,33 @@ private async Task CheckTenureInvestigation(StateMachine<string, string>.Transit
             await TriggerStateMachine(processRequest).ConfigureAwait(false);
         }
 
+        private async Task CheckHOApproval(StateMachine<string, string>.Transition transition)
+        {
+            var processRequest = transition.Parameters[0] as ProcessTrigger;
+            var formData = processRequest.FormData;
+
+            var expectedFormDataKeys = new List<string> { SoleToJointFormDataKeys.HORecommendation };
+            SoleToJointHelpers.ValidateFormData(formData, expectedFormDataKeys);
+            var housingOfficerRecommendation = formData[SoleToJointFormDataKeys.HORecommendation].ToString();
+
+            switch (housingOfficerRecommendation)
+            {
+
+                case SoleToJointFormDataValues.Approve:
+                    processRequest.Trigger = SoleToJointInternalTriggers.HOApprovalPassed;
+                    break;
+                case SoleToJointFormDataValues.Decline:
+                    processRequest.Trigger = SoleToJointInternalTriggers.HOApprovalFailed;
+                    break;
+                default:
+                    throw new FormDataInvalidException(String.Format("Housing Officer Recommendation must be one of: [{0}, {1}] but the value provided was: '{2}'.",
```

Comment:
Do you think we should make another exception for FormDataValueInvalid? Can also be used in the tenure investigation recommendation step